packages/kokkos/CMakeLists.txt

```diff
@@ -280,21 +279,6 @@ IF (KOKKOS_HAS_TRILINOS)
     SET(KOKKOSCORE_XCOMPILER_OPTIONS "${KOKKOSCORE_XCOMPILER_OPTIONS} -Xcompiler ${XCOMP_FLAG}")
     LIST(APPEND KOKKOS_ALL_COMPILE_OPTIONS -Xcompiler ${XCOMP_FLAG})
   ENDFOREACH()
-  SET(KOKKOSCORE_CXX_FLAGS "${KOKKOSCORE_COMPILE_OPTIONS} ${KOKKOSCORE_XCOMPILER_OPTIONS}")
-  IF (KOKKOS_ENABLE_CUDA)
-    STRING(REPLACE ";" " " KOKKOSCORE_CUDA_OPTIONS    "${KOKKOS_CUDA_OPTIONS}")
-    FOREACH(CUDAFE_FLAG ${KOKKOS_CUDAFE_OPTIONS})
-      SET(KOKKOSCORE_CUDAFE_OPTIONS "${KOKKOSCORE_CUDAFE_OPTIONS} -Xcudafe ${CUDAFE_FLAG}")
-      LIST(APPEND KOKKOS_ALL_COMPILE_OPTIONS -Xcudafe ${CUDAFE_FLAG})
```

Comment:
Removing this block of code seems to have the side effect that the cuda fe flags are no longer passed to Trilinos through KOKKOS_ALL_COMPILE_OPTIONS. May be good to doublecheck if that’s intended.

Reply:
You're right. For now, it's better not to modify `KOKKOS_ALL_COMPILE_OPTIONS`. I pushed a corresponding commit.

Reply:
OK!

Reply:
The handling of `KOKKOS_ALL_COMPILE_OPTIONS` is a bit confusing to me so I don't understand the logic here.  From what I can tell, these options are written to a configured file  `KokkosTrilinosConfig.cmake` and added to the `Kokkos::kokkos` target using `target_compiler_options()`.  Then that file gets read in and written to the end of the generated file `KokkosConfig.cmake` which gets installed.  Why not just directly add those compile options to the individual Kokkos libraries using `target_compiler_options()`?  Then those options would be carried along with the direct Kokkos libraries no matter how they are used.
s

Reply:
I agree but `KOKKOS_ALL_COMPILE_OPTIONS` doesn't cause the issue we are trying to fix here and we wanted to keep the changes here minimal.
Note that all this logic is just for `Trilinos`. A standalone `Kokkos` doesn't even go through these lines of code.

With the refactoring in #11779 we should eliminate all the weirdness we had in Kokkos just to make Trilinos work.